### PR TITLE
fix(rulesets): require node_module

### PR DESCRIPTION
**Pull request by P0lip**
Fixes https://github.com/stoplightio/spectral/issues/1026

**Checklist**

- [x] Tests added / updated
- [x] Docs added / updated

**Does this PR introduce a breaking change?**

- [ ] Yes
- [x] No

### CODE REVIEW

**Comment by nulltoken** [comment]
@P0lip This PR triggers a warning during the building of the Spectral binary:

```
> pkg@4.4.4
> Targets not specified. Assuming:
  node10-linux-x64
> Fetching base Node.js binaries to PKG_CACHE_PATH
  fetched-v10.17.0-linux-x64   [] 0%  fetched-v10.17.0-linux-x64   [] 100%
> Warning Malformed requirement for 'p'
  /home/circleci/project/dist/rulesets/evaluators.js
```

File: test-harness/scenarios/require-module.scenario [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+fs.mkdirSync(dir, { recursive: true });
+fs.writeFileSync(path.join(dir, 'package.json'), JSON.stringify({ main: './index.js' }));
+fs.writeFileSync(path.join(dir, 'index.js'), 'module.exports = Function()');
+
+try {
+  require('foo'); // this will throw if foo cannot be resolved
+} finally {
+  fs.unlinkSync(path.join(dir, 'package.json'));
+  fs.unlinkSync(path.join(dir, 'index.js'));
+  fs.rmdirSync(dir);
```
Comment (nulltoken) [inline test-harness/scenarios/require-module.scenario]: @P0lip This doesn't seem to work properly. After the run, a `/tmp/node_modules` folder still exists

**Comment by nulltoken** [comment]
@P0lip Regarding the warning that pkg spits out, shouldn't we find a way to fail the build when that happens?

**Comment by P0lip** [comment]
> @P0lip This PR triggers a warning during the building of the Spectral binary:
> @P0lip Regarding the warning that pkg spits out, shouldn't we find a way to fail the build when that happens?

Yeah, same on my end. Shouldn't be harmful though. It's due to that `require.resolve` call, but everything works fine.

File: test-harness/scenarios/require-module.scenario (hunk 1 above)
Comment (P0lip) [reply test-harness/scenarios/require-module.scenario]: Hmm, good catch. I'll take a look

**Comment by nulltoken** [comment]
> Yeah, same on my end. Shouldn't be harmful though. It's due to that `require.resolve` call, but everything works fine.

So your thoughts are: "If the test harness works ok, then the binary should have been properly packaged"? Good enough for me.

**Comment by nulltoken** [comment]
@P0lip LGTM

**Comment by P0lip** [comment]
> So your thoughts are: "If the test harness works ok, then the binary should have been properly packaged"? Good enough for me.

I tested the binary manually as well.

**Pull request merged**